Search-algorithms/src/base/QuicksortV1.java

```diff
@@ -24,30 +24,18 @@ private Integer partition(Integer[] currentArray, Integer first, Integer last){
 		currentArray=swap(pivotIndex+1,last,currentArray);
 		return pivotIndex+1;
 	}
-	
-
 	@Override
 	public Integer[] sorting(Integer[] arrayToBeOrdered, Integer firstElement, Integer lastElement) {
 		 
 		if(lastElement>firstElement){
-			Integer index=partition(arrayToBeOrdered,firstElement,lastElement);
+			Integer index=partition(arrayToBeOrdered,firstElement,lastElement-1);
```

Comment:
non ricordo se qui doveva starci lastElement o lastElement-1 ricontrolla, non si sa mai

Reply:
va bene lastElement-1 perchè il metodo sorting riceven in ingresso "v.lenght" che di per se è un valore che non ce nel vettore( a voce ti spiego meglio )
siccome partition vuole l'indice dell'ultimo elemento che sarà il pivot devo passarli quindi lastElement-1